streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -386,7 +385,7 @@ public void closeTopology() {
     public void close() {
         log.debug("{} Closing processor topology", logPrefix);
 
-        this.partitionGroup.close();
+        partitionGroup.close();
```

Comment:
+1 to what @ijuma mentioned in https://github.com/apache/kafka/pull/2773: we can close the producer inside task.close if it does has one. And hence `closeProducer` could just be private and we do not need to guard potential bugs either.